ZeekPluginDynamic.cmake

```diff
@@ -147,12 +147,36 @@ function (zeek_add_dynamic_plugin ns name)
     set(dist_tarball_path ${CMAKE_CURRENT_BINARY_DIR}/${dist_tarball_name})
     message(STATUS "Install prefix for plugin ${canon_name}: ${install_dir}")
     message(STATUS "Tarball path for plugin ${canon_name}: ${dist_tarball_path}")
+
+    # Handle dist files: Pre-place user provided dist files into the
+    # directory that zeek-plugin-create-package.sh will tar up. Previously,
+    # the zeek-plugin-create-package.sh script had done this itself.
+    #
+    # Supports only relative paths (relative to CMAKE_CURRENT_SOURCE_DIR) and
+    # strips any leading ../ sequences when determining the location in dist/.
+    set(dist_tarball_dir ${CMAKE_CURRENT_BINARY_DIR}/dist/${canon_name})
+
+    add_custom_target(${target_name}_dist_files)
+    foreach (df ${FN_ARGS_DIST_FILES})
+        set(df_src ${CMAKE_CURRENT_SOURCE_DIR}/${df})
+        if (NOT EXISTS "${df_src}")
+            # This was silently ignored previously, start to warn about it.
+            message(WARNING "The file ${df} (${df_src}) given to DIST_FILES does not exist")
+            continue()
+        endif ()
+
+        string(REGEX REPLACE "^\.\./+" "" df_dist ${df})
+        add_custom_command(
+            TARGET ${target_name}_dist_files
+            COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/${df}
+                    ${dist_tarball_dir}/${df_dist})
+    endforeach ()
```

Comment:
> Is this more in the direction you were thinking of?

Definitely, I am surprised there is already a `DIST_FILES` argument we could use, and it seems to be at a useful level of abstraction.

On a practical level, for the implementation I would still try to get rid of the magic treatment of `..` and instead e.g., simply state that files are placed at the same place in the dist tarball as they were in in the CMake project of the package regardless of where the function call was (e.g., a file `<root>/README` would be placed at `<dist>/README`). I'd also completely get rid of the magic treatment of `..` which assumes something about where this is invoked, maybe something like this:

```suggestion
    add_custom_target(${target_name}_dist_files)
    foreach (df ${FN_ARGS_DIST_FILES})
        # Compute absolute paths for each dist file.
        # NOTE: This is a noop if the path is already absolute.
        get_filename_component(df_src "${df}" RELATIVE BASE_DIR "${CMAKE_SOURCE_DIR}")
        if (NOT EXISTS "${df_src}")
            # This was silently ignored previously, start to warn about it.
            message(WARNING "The file ${df} (${df_src}) given to DIST_FILES does not exist")
            continue()
        endif ()

        # Compute target path in the dist tarball.
        # `file(RELATIVE_PATH)` requires an absolute path.
        file(RELATIVE_PATH df_dist ${CMAKE_SOURCE_DIR} "${df_src}")

        # FIXME(bbannier): Unsure whether this will silently override files or not. Ideally we wouldn't override existing files.
        add_custom_command(
            TARGET ${target_name}_dist_files
            COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/${df}
                    ${dist_tarball_dir}/${df_dist})
    endforeach ()

```